torch/_inductor/lowering.py

```diff
@@ -1116,6 +1119,20 @@ def randn(*args, **kwargs):
         return fast_randn(*args, **kwargs)
 
 
+@register_lowering([aten.randn_like])
+def randn_like(x, **kwargs):
+    if config.fallback_random:
+        return fallback_randn_like(*args, **kwargs)
+    else:
+        dtype = kwargs.pop("dtype", None)
+        dtype = x.get_dtype() if dtype is None else dtype
+
+        device = kwargs.pop("device", None)
+        device = x.get_device() if device is None else device
+
+        return fast_randn(x.get_size(), dtype=dtype, device=device, **kwargs)
```

Comment:
hm, this won't preserve permutation if `x` is permuted?

Reply:
it won't, but the MO thus far is we only set permutation requirements on graph outputs